src/test/scala/mesosphere/marathon/MarathonSchedulerServiceTest.scala

```diff
@@ -194,6 +189,8 @@ class MarathonSchedulerServiceTest
       override def newTimer() = mockTimer
     }
 
+    schedulerService.timer = mockTimer
```

Comment:
No, the test relies on the timer instance being the same – if the `newTimer` is not overridden, a new timer will be created (instead of the mock).

Not a very nice test, but I resisted refactoring everything ...